### ISSUE TITLE
Paste from clipboard and some important requests !

### ISSUE THREAD

**Issue by trimechee** [database64128/youtube-dl-wpf#131]
Hello ,I discover and use now Cube YouTube Downloader - youtube-dl-wpf and it is absolutely sensational and wonderful work of art ! Thank you so much ! 

I have a some requests for even more improvement please: manuel and automatic paste url adress from clipboard and and merge and unify the paste from clipboard button and download button to make only one button click and thus=a faster comfortable download,  batch downloader ti add multiple url adress to the list and download them later, clearer warning to let us know the end of the download and its success, and an option to exit or shut down the computer after download finish, and to facilitate and simplify the task for beginners and ordinary users who know nothing about computers because finding the right compatible yt-dlp/FFmpeg version can be difficult, that's why it's good idea to make a classic installer for windows 10 which allows us to automatically install yt-dlp/FFmpeg and update them , and add preset sponsorblock to default preset and add sponsorblock to worst audio preset or make new preset worst audio sponsorblock because il may be difficlut to newbies to find and add preset and i would like to recommend this software to some of my friends and twitter followers who know very little things about computers adn softwares option.... thank you very much !

and I contacted the popular and famous site "videohelp" a few days ago to add your great and awesome software !

https://www.videohelp.com/software/Cube-YouTube-Downloader-

**Comment by database64128** [comment]
First of all, thank you for your support! I did notice a lot of traffic coming from videohelp and was curious who submitted it. Now that I know, I want you to know that I really appreciate your efforts.

Unfortunately, I have to say no to some of your requests. `youtube-dl-wpf` is just a small side project of mine. My approach to its development will always be to prioritize my own use cases, while also catering to the community as much as I can. As you may have noticed, I have already implemented a bunch of new features and made quite a few changes based on community feedback. Decisions on whether to adopt a proposal from the community are made by evaluating the usefulness of the proposed feature or change for me and the wider community, and the feasibility of implementing it in my limited spare time.

Now that I have explained my ways of doing side projects to you, let's go through your proposals:

> One button for pasting URL from the clipboard and start the download.

I always find the practice of an app actively reading the clipboard content to be kind of invasive. The only way for me to give you the content of my clipboard should be me pressing Ctrl + V. Right now you can already press Ctrl + V and hit Enter to quickly start a download. In my opinion it's even faster and more convenient than moving your mouse cursor to the button and clicking it.

> Option to shut down the computer after finishing download tasks.

I don't feel comfortable implementing something that can do stuff like shutting down the whole system. Personally I never shut down my computers. Whenever I want to use them, they are always ready for use.

> Make it easier for non-technical folks to use.

While I believe good UX is something that empowers everyone who uses it, this project at its current state is certainly not for someone without the necessary technical skills. It takes a lot of time and effort to improve UX, and I'm not really interested in it.

> Add presets with sponsorblock.

Now this does look like something I could implement. After looking into it, I feel it would be better to implement it as an option, like the custom download path. It could give you some builtin values like `default`, `all`, `intro`, `outro` to select from, and also allow you to specify custom values. The option would generate `--sponsorblock-remove <arg>`. What do you think?

**Comment by trimechee** [comment]
Yes! adding "--sponsorblock-remove <arg>" will be a great brilliant option needed because a lot of youtube videos include ads !!

Thank you very much for your valuable advice and recommendations and Cube YouTube Downloader - youtube-dl-wpf is already complete and extraordinary!

yes I understand that it's tricky to turn off the computer, so maybe it will be good to add a visual or sound warning to warn us of the end of the download when we download a lot of large videos....

It will also be great to add a big Cube YouTube Downloader sponsor button on the main topic slot to encourage you and thank you for continuing to develop this brilliant project, so people can donate via paypal, bitcoin.... or add discreet non-intrusive ads and for people who don't like ads, there would be an option to disable ads...


please i have an important request , can you add support for this rare site , because i found this rare exceptional sound which is the amazing combination of White Noise + Rain which cancels the ambient noise, i tried to download using yt dlp and video extensions like videodownloadhlper but i did not succeed, i did not I only managed to download a little piece of 30 seconds sound.....
can our beloved Cube YouTube Downloader allow us to download this rare sound please? I just want to download this sound and not the other sounds on the site, this sound is free but I still have to be connected to the internet that's why I want to download it for offline use....thank you very much :)

https://mynoise.net/NoiseMachines/whiteRainNoiseGenerator.php

**Comment by database64128** [comment]
My guess is it's just playing that 30s clip in a loop.

**Comment by trimechee** [comment]
I did a search, there are some white noise that make noise in a loop but scientists say it's better if white noise is a continuous natural sound without effect in a loop, and this "mynoise" website is managed by a great famous scientist who recorded this sound in the plains of Indonesia so i don't think it's in a loop noise, i think the sound is protected and we can't download it the classic way, apparently we have to download every bit of 30 seconds via "video downloadhelper" extension and find a program to put them together that's why i made my request.....
